src/zdir.c

```diff
@@ -105,9 +105,15 @@ zdir_new (const char *path, const char *parent)
 {
     zdir_t *self = (zdir_t *) zmalloc (sizeof (zdir_t));
     assert (self);
+    assert (path);
+    assert (parent);
```

Comment:
not sure assert (parent) followed by if (parent) makes sense?

Reply:
I think the `if parent` comes from a previous life? So far i'm puzzled that some tests apparently passed the null balue (so are broken now) and saw no docu-comments to specify if this input can be assumed valid with such and such behavior then.

Reply:
I don't know, haven't really used this class before so can't say, sorry